Zend/zend_opcode.c

```diff
@@ -367,12 +367,13 @@ ZEND_API void destroy_op_array(zend_op_array *op_array)
 		zend_llist_apply_with_argument(&zend_extensions, (llist_apply_with_arg_func_t) zend_extension_op_array_dtor_handler, op_array);
 	}
 	if (op_array->arg_info) {
-		uint32_t num_args = op_array->num_args;
+		int32_t num_args = op_array->num_args;
+		int32_t i = op_array->fn_flags & ZEND_ACC_HAS_RETURN_TYPE ? -1 : 0;
 
 		if (op_array->fn_flags & ZEND_ACC_VARIADIC) {
 			num_args++;
 		}
-		for (i = 0; i < num_args; i++) {
+		for ( ; i < num_args; i++) {
 			zend_string_release(op_array->arg_info[i].name);
```

Comment:
I also didn't see problems running 'make test'.
I found problems only when I ran tests with valgrind and only on few tests.
Travis was lucky to crash on each test. Thanks to it :)

Dmitry.

On Wed, Jan 28, 2015 at 10:08 AM, Levi Morrison <notifications@github.com>
wrote:

> In Zend/zend_opcode.c
> <https://github.com/php/php-src/pull/997#discussion_r23669185>:
>
> >
> >  		if (op_array->fn_flags & ZEND_ACC_VARIADIC) {
> >  			num_args++;
> >  		}
> > -		for (i = 0; i < num_args; i++) {
> > +		for ( ; i < num_args; i++) {
> >  			zend_string_release(op_array->arg_info[i].name);
>
> Note that I made a change here because I was previously getting memory
> leaks. Since this fixed the leaks and had no issues on debug or release
> builds I pushed for others to test. I'm still confused why I had no issues
> with tests. All in all, glad the root cause was fixed. Thanks Dmitry.
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/php/php-src/pull/997/files#r23669185>.
>